support-frontend/stories/checkouts/CheckoutTopUpAmounts.stories.tsx

```diff
@@ -23,39 +23,36 @@ export default {
 function Template(args: CheckoutTopUpAmountsProps) {
 	return (
 		<CheckoutTopUpAmounts
+			contributionType="MONTHLY"
 			currencySymbol={args.currencySymbol}
-			timePeriod={args.timePeriod}
-			amounts={args.amounts}
 			isWithinThreshold={args.isWithinThreshold}
 		/>
 	);
 }
 
-Template.args = {} as Record<string, unknown>;
+Template.args = {} as CheckoutTopUpAmountsProps;
 
 export const UKMonthlyTopUpAmounts = Template.bind({});
 
 UKMonthlyTopUpAmounts.args = {
+	contributionType: 'MONTHLY',
 	currencySymbol: '£',
 	timePeriod: 'month',
 	amounts: [1, 2, 3, 4, 5],
 	isWithinThreshold: true,
-};
+} as CheckoutTopUpAmountsProps;
```

Comment:
Would be good to avoid these type assertions if possible- what are the issues that are causing their use?

Reply:
The issue I was having is that when setting `ComponentNameThatInheritsTemplate.args` they weren't being typed, so they were a free for all.

What's the benefit of creating the `Template` abstraction vs 
```tsx
const UKMonthlyTopUpAmounts = () => {
  return <CheckoutTopUpAmounts contributionType={"MONTHLY"} /* ... */ />
}

const UKYearlyTopUpAmounts = () => {
  return <CheckoutTopUpAmounts contributionType={"YEARLY"} /* ... */ />
}
```
?

Reply:
The Template abstraction is what's demonstrated in the Storybook docs and as far as I'm aware is part of how its built-in documentation and controls features work- as far as my understanding goes your suggestion would destroy the automatic detection of prop types and require us to add this stuff manually for every story.

Asserting the type of the `args` property on `Template` should propagate that type to every story that derives from `Template`- this seems to be working fine when looking at other stories. If `IndividualStory.args` isn't correctly typed then I'd imagine something else is wrong somewhere and using a type assertion is simply hiding the real issue.

Reply:
This all seems to work as you've described, so perhaps something else got mixed up in my IDE/head.

Dropping [this link](https://storybook.js.org/tutorials/intro-to-storybook/ember/en/simple-component/) here for any future folk who fancy reading about it. Thanks for the info!